include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -185,304 +185,253 @@ class __kernel_name_base
     {
 #if _ONEDPL_KERNEL_BUNDLE_PRESENT
         auto __kernel_bundle = sycl::get_kernel_bundle<sycl::bundle_state::executable>(__exec.queue().get_context());
-        return __kernel_bundle.get_kernel(sycl::get_kernel_id<_DerivedKernelName>());
+        return __kernel_bundle.get_kernel(sycl::get_kernel_id<_DerivedKernelName...>());
```

Comment:
As far as I got the main problem for unnamed_lambda mode when the pack `_DerivedKernelName..  is empty....
And we cannot use pre-compiled kernel mechanism(to get work group size) at all.
Interesting what's compiler team answer give here.

Reply:
Seems like workaround is found. We decided to use mixed approach and compile kernels only where we need reliable work group size. Otherwise, use the solution with optional kernel name with help of variadic templates.